### ISSUE TITLE
Suggestion: "Smart Tab" behaviour

### ISSUE THREAD

**Issue by Alhadis** [atom/atom#10986]
If somebody asked me to imagine a perfect world, it'd be one where coders used hard tabs for indentation, and spaces to align all code after a non-whitespace character.

While this world will never exist, it does illustrate a potentially useful addition to Atom: a fourth tab-type option, which I'm calling _"smart tab"_ for the purpose of this issue:
<img width="173" alt="screen shot 2016-02-27 at 6 00 05 pm" src="https://cloud.githubusercontent.com/assets/2346707/13371508/03162026-dd7c-11e5-9a35-617ae731a436.png">

Basically, the editor switches between hard tabs for indentation and spaces depending on whether or not there's a non-whitespace character before the cursor.
<img height="110" src="https://cloud.githubusercontent.com/assets/2346707/13371554/4118b14e-dd7d-11e5-80bd-f2ee597aa01f.png">
